waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/DefaultMetaStoreClientFactory.java

```diff
@@ -41,6 +43,8 @@ private static class ReconnectingMetastoreClientInvocationHandler implements Inv
     private final String name;
     private final int maxRetries;
 
+    private HiveUgiArgs cachedUgi = HiveUgiArgs.WAGGLE_DANCE_DEFAULT;
```

Comment:
Setting the default so it's always set.
This can happen if Waggle Dance for instance makes a client and does call itself (via our status/health checks).
If any other Client (Trino/Presto/Qubole/Hive/Spark) might have a different lifecycle than we can trace this user back and we'll know from Metastore logs the call originated from Waggle Dance.

Reply:
That's a great idea, love it.